src/XRP/model/xrp-currency-amount.ts

```diff
@@ -14,31 +15,32 @@ export default class XrpCurrencyAmount {
    * @returns an XrpCurrencyAmount with its fields set via the analogous protobuf fields.
    * @see https://github.com/ripple/rippled/blob/develop/src/ripple/proto/org/xrpl/rpc/v1/amount.proto#L10
    */
-  public static from(
-    currencyAmount: CurrencyAmount,
-  ): XrpCurrencyAmount | undefined {
-    switch (currencyAmount.getAmountCase()) {
-      // Mutually exclusive: either drops or issuedCurrency is set in an XRPCurrencyAmount
-      case CurrencyAmount.AmountCase.ISSUED_CURRENCY_AMOUNT: {
-        const issuedCurrencyAmount = currencyAmount.getIssuedCurrencyAmount()
-        if (issuedCurrencyAmount) {
-          const issuedCurrency = XrpIssuedCurrency.from(issuedCurrencyAmount)
-          if (issuedCurrency) {
-            return new XrpCurrencyAmount(undefined, issuedCurrency)
-          }
-        }
-        return undefined
+  public static from(currencyAmount: CurrencyAmount): XrpCurrencyAmount {
+    // Mutually exclusive: either drops or issuedCurrency is set in an XRPCurrencyAmount
+    if (
+      currencyAmount.hasIssuedCurrencyAmount() &&
+      !currencyAmount.hasXrpAmount
```

Comment:
what if you did

```
const issuedCurrencyAmount = currencyAmount.getIssuedCurrencyAmount()
const xrpAmount = currencyAmount.getXrpAmount()
```
and then used their truthiness in your iff statements? then typescript will also know whether they're truthy or not